src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Android.Accounts;
+using Android.App;
+using Android.Content;
+using AndroidNative = Android;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal.Broker;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.UI;
+using Microsoft.Identity.Client.Utils;
+using System.Globalization;
+using Microsoft.Identity.Json.Linq;
+using Microsoft.Identity.Json;
+using Android.OS;
+
+namespace Microsoft.Identity.Client.Platforms.Android
+{
+    [AndroidNative.Runtime.Preserve(AllMembers = true)]
+    internal class AndroidBroker : IBroker
+    {
+        private static SemaphoreSlim readyForResponse = null;
+        private static MsalTokenResponse resultEx = null;
```

Comment:
Why does the response have to be static?

Reply:
this does not

Reply:
The broker result is set by a callback from AuthenticationContinuationHelper which does not have a reference to the broker object. Because of this, the result needs to be static so that the broker object can pick it up and process it. If you have a better way of resolving this issue we should discuss.